internal/auth/token_manager.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2025 Ping Identity Corporation
+
+package auth_internal
+
+import (
+	"fmt"
+
+	"github.com/pingidentity/pingcli/internal/configuration/options"
+	"github.com/pingidentity/pingcli/internal/errs"
+	"github.com/pingidentity/pingcli/internal/profiles"
+	"github.com/pingidentity/pingone-go-client/config"
+	svcOAuth2 "github.com/pingidentity/pingone-go-client/oauth2"
+	"golang.org/x/oauth2"
+)
+
+// TokenManager defines the interface for managing OAuth2 tokens in the keychain
+type TokenManager interface {
+	SaveToken(token *oauth2.Token) error
+	LoadToken() (*oauth2.Token, error)
+	ClearToken() error
+	HasToken() bool
+}
+
+// DefaultTokenManager implements the TokenManager interface using the default pingcli keychain service
+type DefaultTokenManager struct {
+	serviceName string
+}
+
+// NewDefaultTokenManager creates a new DefaultTokenManager instance
+func NewDefaultTokenManager() TokenManager {
+	return &DefaultTokenManager{
+		serviceName: "pingcli",
+	}
+}
+
+// GetCurrentAuthMethod returns the configured authentication method key for the active profile
+func GetCurrentAuthMethod() (string, error) {
+	authMethod, err := profiles.GetOptionValue(options.PingOneAuthenticationTypeOption)
+	if err != nil {
+		return "", fmt.Errorf("failed to get current auth method: %w", err)
+	}
+
+	if authMethod == "" {
+		return "", ErrAuthMethodNotConfigured
+	}
+
+	return GetAuthMethodKey(authMethod)
+}
+
+// GetAuthMethodKey generates a unique keychain account name for the given authentication method
+// using the environment ID and client ID from the profile configuration
+func GetAuthMethodKey(authMethod string) (string, error) {
```

Comment:
If a user changes their client ID in their configuration for example, will they be left forever with remnants in their keychain that the logout tool won't clear? I think we need to be sure that logout handles tokens that may have come from an older version of the user's config.